### PR TITLE
Lmdevs 278 add accounting tests

### PR DESCRIPTION
**Pull request by dospore**
### Motivation
- Add tests for position account functions to verify the information being displayed is correct

### Changes
- Add basic positions tests, these are standard short and long positions
- Add underwater position tests, these are accounts that can be liquidated
- Updated calcProfitableLiquidationPrice to match base and quote changes
- Update calcBorrowed to match base and quote changes

### CODE REVIEW

File: src/Accounting/index.ts [hunk 1]
```diff
@@ -58,16 +61,12 @@ export const calcProfitableLiquidationPrice: (
...
-                  (quote - maxLeverage * quote)
-            : 0 + quote < 0 // case 2
-            ? -1 *
-              ((base * (maxLeverage - (RYAN_6 * LIQUIDATION_GAS_COST - LIQUIDATION_GAS_COST) * maxLeverage)) /
-                  (maxLeverage * quote + quote))
+    if (borrowed > 0 || base < 0) {
+        return base > 0 // case 1
+            ? (maxLeverage * (quote - (RYAN_6 * LIQUIDATION_GAS_COST - LIQUIDATION_GAS_COST))) / (base - maxLeverage * base)
+            : 0 + base < 0 // case 2
+            ? (-1 * (quote * maxLeverage - (RYAN_6 * LIQUIDATION_GAS_COST - LIQUIDATION_GAS_COST) * maxLeverage)) / (maxLeverage * base + base)
```
Comment (BenjaminPatch) [inline src/Accounting/index.ts:69]: Can you define these in variables, so we know what they are supposed to represent please :100:

**Review by BenjaminPatch — CHANGES_REQUESTED** [review]

File: test/accountingTests.ts [hunk 2]
```diff
@@ -1,5 +1,38 @@
...
+  calcLiquidationPrice,
+  calcProfitableLiquidationPrice,
+} from "../src/Accounting"
+
+
+// Basic positions
+// The user holds a position not in liquidation
+// Deposit 200 at $100, short 10 units
+const position1 = {
+  quote: 1200, base: -10, price: 100, maxLeverage: 50
```
Comment (raymogg) [inline test/accountingTests.ts:19]: Do you think it matters we don't use WAD values for these? Will just need to be careful when we move to using this with real data. The math should hold. although we may need to use BN in that case

File: src/Accounting/index.ts [hunk 3]
```diff
@@ -58,16 +61,12 @@ export const calcProfitableLiquidationPrice: (
...
-        return quote > 0 // case 1
-            ? (maxLeverage * (base - (RYAN_6 * LIQUIDATION_GAS_COST - LIQUIDATION_GAS_COST))) /
-                  (quote - maxLeverage * quote)
-            : 0 + quote < 0 // case 2
-            ? -1 *
-              ((base * (maxLeverage - (RYAN_6 * LIQUIDATION_GAS_COST - LIQUIDATION_GAS_COST) * maxLeverage)) /
-                  (maxLeverage * quote + quote))
+    if (borrowed > 0 || base < 0) {
+        return base > 0 // case 1
+            ? (maxLeverage * (quote - (RYAN_6 * LIQUIDATION_GAS_COST - LIQUIDATION_GAS_COST))) / (base - maxLeverage * base)
```
[the remark this answers is not in the document]
Comment (raymogg) [reply src/Accounting/index.ts:67]: Dev corner sentiment: please change this block

File: src/Accounting/index.ts (hunk 1 above)
Comment (dospore) [reply src/Accounting/index.ts:69]: which ones the 0's and the -1??

File: src/Accounting/index.ts [hunk 4]
```diff
@@ -6,9 +6,12 @@ const LIQUIDATION_GAS_COST = 25; // When gas price is 250 gwei and eth price is
 
 /**
  * Calculates the leverage multiplier the user currently has based on their position and a given price
+ * @returns the leverage of the user at a given position or -1 if the position is invalid this represents infinite leverage
  */
 export const calcLeverage: (quote: number, base: number, price: number) => number = (quote, base, price) => {
-    return calcNotionalValue(base, price) / totalMargin(base, quote, price);
+    const margin = calcTotalMargin(quote, base, price);
+    if (margin <= 0) return -1
+    return calcNotionalValue(base, price) / calcTotalMargin(quote, base, price);
```
[the remark this answers is not in the document]
Comment (dospore) [reply src/Accounting/index.ts:14]: doneski

File: src/Accounting/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (dospore) [reply src/Accounting/index.ts:67]: lame

File: test/accountingTests.ts (hunk 2 above)
Comment (dospore) [reply test/accountingTests.ts:19]: yeah I was thinking about this, its probably better if I use WAD

File: src/Accounting/index.ts (hunk 1 above)
Comment (BenjaminPatch) [reply src/Accounting/index.ts]: The whole things. Like what does the numerator represent, and the denominator? I get that these are the formulas for calculating the liquidation price, but anybody who hasn't been able to go through the accounting doc is going to have a really hard time figuring out how this works, nothing is named

**Review by raymogg — APPROVED** [review]

**Pull request merged**